common/logf.c

```diff
@@ -268,6 +273,105 @@ logf_handler_set_prio(logf_handler_t *handler, logf_prio_t prio)
 	handler->prio = prio;
 }
 
+static int
+logf_current_log_cb(const char *path, const char *file, void *data)
+{
+	list_t **log_file_list = (list_t **)data;
+
+	str_t *path_str = str_new(path);
+	str_append(path_str, "/");
+	str_append(path_str, file);
+
+	//only process .current symlinks.
+	if ((strstr(file, ".current") == NULL) || (!file_is_link(str_buffer(path_str)))) {
+		return 0;
+	}
+
+	char buffer[PATH_MAX];
+
+	//-1 is used to leave space for null terminator.
+	ssize_t len = readlink(str_buffer(path_str), buffer, PATH_MAX - 1);
+
+	if (len != -1) {
+		// Ensure the buffer is null-terminated
+		buffer[len] = '\0';
+		DEBUG("Adding currently used logfile to list %s", buffer);
+		*log_file_list = list_append(*log_file_list, mem_strdup(buffer));
```

Comment:
the data pointer of the list element is never freed. In control.c only list_delete() is called
which frees meta data only of a list.
provide a helper logf_log_files_list_free() which does a deep free and call this in control.c

Reply:
sounds good, we should do this in a separate PR adapting existing locations where applicable